backend/src/scripts/utility/teamspaces/removeIncompleteRevisions.js

```diff
@@ -0,0 +1,214 @@
+		{ filename: { $regex: `^/${teamspace}/${model}/revision/${UUIDToString(revId)}/` } },
+	));
+
+	modelStashPromises.push(processFilesAndRefs(
+		teamspace,
+		`${model}.stash.json_mpc.ref`,
+		{ _id: { $regex: `^${UUIDToString(revId)}/` } },
+	));
+
+	const supermeshIds = (await find(teamspace, `${model}.stash.3drepo`, { rev_id: revId, type: 'mesh' }, { _id: 1 })).map((r) => UUIDToString(r._id));
+	supermeshIds.map((supermeshId) => {
+		modelStashPromises.push(processFilesAndRefs(teamspace, `${model}.stash.3drepo.ref`, { _id: { $in: [
+			`${supermeshId}_faces`,
+			`${supermeshId}_normals`,
+			`${supermeshId}_vertices`,
+		] } }));
+		modelStashPromises.push(processFilesAndRefs(teamspace, `${model}.stash.json_mpc.ref`, { _id: { $in: [
+			`${supermeshId}.json.mpc`,
+			`${supermeshId}_unity.json.mpc`,
+		] } }));
+		modelStashPromises.push(processFilesAndRefs(teamspace, `${model}.stash.src.ref`, { _id: { $in: [
+			`${supermeshId}.src.mpc`,
+		] } }));
+		modelStashPromises.push(processFilesAndRefs(teamspace, `${model}.stash.unity3d.ref`, { _id: { $in: [
+			`${supermeshId}.unity3d`,
+		] } }));
+
+		return supermeshId;
+	});
+
+	modelStashPromises.push(removeRecords(teamspace, `${model}.stash.unity3d`, { _id: revId }));
+	modelStashPromises.push(removeRecords(teamspace, `${model}.stash.3drepo`, { rev_id: revId }, '_extRef'));
+
+	await Promise.all(modelStashPromises);
+};
+
+const processModelSequences = async (teamspace, model, revId) => {
+	const sequences = await find(teamspace, `${model}.sequences`, { rev_id: revId }, { frames: 1 });
+	const sequenceFilenames = [];
+
+	for (const { _id, frames } of sequences) {
+		if (frames) {
+			for (const { state } of frames) {
+				sequenceFilenames.push(state);
+			}
+		}
+
+		// eslint-disable-next-line no-await-in-loop
+		await Promise.all([
+			removeRecords(teamspace, `${model}.activities`, { sequenceId: _id }, '_extRef'),
+			processFilesAndRefs(teamspace, `${model}.activities.ref`, { _id: UUIDToString(_id) }),
+		]);
+	}
+
+	await processFilesAndRefs(teamspace, `${model}.sequences.ref`, { _id: { $in: sequenceFilenames } });
+
+	await deleteMany(teamspace, `${model}.sequences`, { rev_id: revId });
+};
+
+const processModelScene = async (teamspace, model, revId) => {
+	await removeRecords(teamspace, `${model}.scene`, { rev_id: revId }, '_extRef');
+};
+
+const removeRevision = async (teamspace, model, revId) => {
+	logger.logInfo(`\t\t-${model}::${UUIDToString(revId)}`);
+
+	await Promise.all([
+		processModelScene(teamspace, model, revId),
+		processModelSequences(teamspace, model, revId),
+		processModelStash(teamspace, model, revId),
+	]);
+};
+
+const processTeamspace = async (teamspace, revisionAge) => {
+	const cols = await getCollectionsEndsWith(teamspace, '.history');
+
+	for (const { name } of cols) {
+		const incompleteRevisionFilter = {
+			incomplete: { $exists: true },
+			timestamp: { $lt: new Date(new Date().setDate(new Date().getDate() - revisionAge)) },
```

Comment:
I don't think you need 3 dates here.... looks a bit mad.

Just instantiate one date and manipulate that